### PR TITLE
Data preservation

### PR DESCRIPTION
**Pull request by rmrmrmr**
## On this pull request @PrangonGhose and I:

- Preserved data for the UI 
- Displayed the information using the preserved data 
- Fixed linter errors

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -22,6 +23,99 @@ def list_books
...
+                        age: person.age,
+                        name: person.name,
+                        classroom: person.classroom,
+                        parent_permission: person.parent_permission
+                      }
+                    end
+      persons_list << persons_obj
+    end
+    File.write('./memory/persons.json', JSON.pretty_generate(persons_list))
+  end
```
Comment (Gambit142) [inline app.rb:69]: - [x] I noticed that when I exit the application, I get the error as shown in the screenshot below and the data is not persisted in the _**JSON**_ files

image

To solve this, you would need to create the directory `memory` before you can create or write in JSON files. Also, the logic for creating the directory `memory` should be that if it exists, it shouldn't be created. 👍 😎

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @rmrmrmr @PrangonGhose 👋, 

Great work so far! I love the UI of the application, especially the way `rentals` are listed for a particular user 👏 

## Highlights
- All linters checks passed ✔️ 
- Pull Request summary is descriptive ✔️
- `Readme.md` file is professional ✔️

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

#### Please, remember to tag me in your question so I can receive the notification. You can tag me by typing @Gambit142

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: app.rb (hunk 1 above)
Comment (rmrmrmr) [reply app.rb:69]: Thanks for the observation and advice. We'll fix it!

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi Team,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**